presto-raptorx/src/main/java/com/facebook/presto/raptorx/metadata/DatabaseMetadata.java

```diff
@@ -0,0 +1,301 @@
+        verifyMetadata(info != null, "Invalid distribution ID: %s", distributionId);
+        return info;
+    }
+
+    @Override
+    public Optional<DistributionInfo> getDistributionInfo(String distributionName)
+    {
+        return Optional.ofNullable(distributionDao.getDistributionInfo(distributionName.getBytes(UTF_8)));
+    }
+
+    @Override
+    public Optional<Long> getSchemaId(long commitId, String schemaName)
+    {
+        return Optional.ofNullable(dao.getSchemaId(commitId, schemaName.getBytes(UTF_8)));
+    }
+
+    @Override
+    public Optional<Long> getTableId(long commitId, long schemaId, String tableName)
+    {
+        return Optional.ofNullable(dao.getTableId(commitId, schemaId, tableName.getBytes(UTF_8)));
+    }
+
+    @Override
+    public Optional<Long> getViewId(long commitId, long schemaId, String viewName)
+    {
+        return Optional.ofNullable(dao.getViewId(commitId, schemaId, viewName.getBytes(UTF_8)));
+    }
+
+    @Override
+    public Collection<String> listSchemas(long commitId)
+    {
+        return dao.listSchemas(commitId).values().stream()
+                .map(bytes -> new String(bytes, UTF_8))
+                .collect(toImmutableList());
+    }
+
+    @Override
+    public Collection<String> listTables(long commitId, long schemaId)
+    {
+        return dao.listTableNames(commitId, schemaId).stream()
+                .map(bytes -> new String(bytes, UTF_8))
+                .collect(toImmutableList());
+    }
+
+    @Override
+    public Collection<String> listViews(long commitId, long schemaId)
+    {
+        return dao.listViewNames(commitId, schemaId).stream()
+                .map(bytes -> new String(bytes, UTF_8))
+                .collect(toImmutableList());
+    }
+
+    @Override
+    public SchemaInfo getSchemaInfo(long commitId, long schemaId)
+    {
+        SchemaInfo info = dao.getSchemaInfo(commitId, schemaId);
+        verifyMetadata(info != null, "Invalid schema ID: %s", schemaId);
+        return info;
+    }
+
+    @Override
+    public TableInfo getTableInfo(long commitId, long tableId)
+    {
+        TableInfo table = dao.getTableInfo(commitId, tableId);
+        verifyMetadata(table != null, "Invalid table ID: %s", tableId);
+
+        List<ColumnInfo> columns = dao.getColumnInfo(commitId, tableId);
+        verifyMetadata(columns != null, "Invalid table ID: %s", tableId);
+        verifyMetadata(!columns.isEmpty(), "No columns for table ID: %s", tableId);
+
+        return table.builder()
+                .setColumns(columns)
+                .build();
+    }
+
+    @Override
+    public ViewInfo getViewInfo(long commitId, long viewId)
+    {
+        ViewInfo view = dao.getViewInfo(commitId, viewId);
+        verifyMetadata(view != null, "Invalid view ID: %s", viewId);
+        return view;
+    }
+
+    @Override
+    public Collection<TableStats> listTableStats(long commitId, Optional<Long> schemaId, Optional<Long> tableId)
```

Comment:
Since tableSizes doesn't have any data structures in Transaction.java.
If t1 is like:
insert 100 chunks;
listTableStats;
t1 commit;

listTableStats doesn't show what t1 does. Only after t1 commits the 100 chunks, listTableStats can show that.

I wouldn't say it's an error. Just wanna confirm it's what we plan to do.

Reply:
This is a good observation, but it's explicitly checked for and not supported. See the first line of `Transaction.listTableStats()` from which this method is called:
```java
if (!actions.isEmpty()) {
    throw new PrestoException(NOT_SUPPORTED, "Table stats cannot be accessed with uncommitted modifications");
}
```
Basically, I was lazy and decided that having the `system.table_stats` table not work inside a transaction would not be a problem in practice. We can implement this later if needed.

Reply:
Note that we would consider this a bug if we allowed access to stale data, as the general policy in Presto is to either return a correct answer or fail the operation.